v2-migration/src/main/resources/META-INF/rewrite/aws-sdk-java-v1-to-v2-with-tm.yml

```diff
@@ -0,0 +1,45 @@
+#
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#  http://aws.amazon.com/apache2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+#
+---
+type: specs.openrewrite.org/v1beta/recipe
+name: software.amazon.awssdk.v2migration.AwsSdkJavaV1ToV2WithTransferManager
+displayName: Migrate from the AWS SDK for Java v1 to the AWS SDK for Java v2 including S3 Transfer Manager.
+description: This recipe will apply changes required for migrating from the AWS SDK for Java v1 to the AWS SDK for Java v2, including S3 Transfer Manager.
+tags:
+  - aws
+  - sdk
+recipeList:
+  - software.amazon.awssdk.v2migration.AddTransferManagerDependency
```

Comment:
Can we just use `software.amazon.awssdk.v2migration.AwsSdkJavaV1ToV2`?

Reply:
Might be easier to have a separate composite recipe, to reference in internal testing. Plus, we may need to add recipes in the middle of the order

Reply:
I see, my concern is that since we updated our tests to test with this new recipe, we won't be able to catch regressions for the main recipe and this diverge would make it harder to maintain in the long term. I wonder if there's a way to add a configuration to the main recipe to toggle the behavior of enabling TM recipe

Reply:
> my concern is that since we updated our tests to test with this new recipe, we won't be able to catch regressions for the main recipe and this diverge would make it harder to maintain in the long term

The TM recipes only add functionality, it shouldn't affect any existing core transforms. The goal is to have one single parent recipe eventually once the TM recipes are complete. 

The current TM transforms make use of V1BuilderVariationsToV2Builder, NewClassToBuilder, and V1SetterToV2, may be more efficient to re-use the logic instead of copying them to new classes. TM code will include S3 POJOs so will need the S3 recipes as well

Reply:
Updated end to end tests to have separate testing for AwsSdkJavaV1ToV2 and AwsSdkJavaV1ToV2WithTransferManager

Reply:
I think creating separate recipes would make it easier to maintain and extend IMO. For example, if we want to support migrating individual service client, we'd need to have separate recipes for those anyway.  We can probably create common utilities to reuse code to reduce duplication.  

With that said, this is not a blocker, but let's add a backlog item to separate them

Reply:
Makes sense, creating tracking ticket for refactor